File: cpp/potega.cpp
/*
 * potega.cpp
 * a0 = 1
 * a1 = a
 * an = a*...*a(n-czynników) dla n zaw. N+ - {1}
 */


#include <iostream>

using namespace std;
 
float potega_it(float x, int n) {
    float wynik = 1;
    for(int i = 0; i< n; i++) {
        cout << i << endl;
        wynik = wynik*i;
    }
    return wynik;
}


int main(int argc, char **argv)
{
	// zadeklaruj zmienne
    // pobieranie od uzytkowanika podstawy i wykładnika
    float x = 0;
    int n = 0;
    cout << "Podaj podstawę: " << endl;
    cin >> x;
    cout << "Podaj wykładnik: " << endl;
    cin >> n;
    cout << "Potęga" << potega_it(x ,n) << endl;
    
	return 0;
}

File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
#


def potega_it(x,n):

    i = 1
    wynik = 1
    while i <= n:
        i = i + i

    return wynik

# potega_rek(a, 0) = 1  dla a różnego od 0
# potega_rek(a,n ) = potega_rek(a, n-1) * a dla n = N+

def potega_rek(a, n):
    if n == 0:
        return 1
    return potega_rek(a, n-1) * a



def main(args):
    x = float(input("Podaj podstawę"))
    n = int(input("Podaj wykładnik potęgi"))
    assert potega_it(1,2) == 1
    assert potega_it(2,2) == 4
    assert potega_it(3,3) == 27
    assert potega_it(4,2) == 16
    print("Wynik:")
    print("Potega:")

    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
